activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -871,9 +871,20 @@ def configure_connection
         end
 
         def column_definitions(table_name) # :nodoc:
-          execute_and_free("SHOW FULL FIELDS FROM #{quote_table_name(table_name)}", "SCHEMA") do |result|
+          results = execute_and_free("SHOW FULL FIELDS FROM #{quote_table_name(table_name)}", "SCHEMA") do |result|
             each_hash(result)
           end
+
+
+          results.tap do |column_infos|
+            if !mariadb? && get_database_version > "8.0.2"
+              column_infos.each do |column_info|
+                if column_info[:Type].match?("binary") && column_info.fetch(:Default)&.start_with?("0x")
```

Comment:
``` suggestion
                if column_info[:Type].include?("binary") && column_info.fetch(:Default)&.start_with?("0x")
```

I think `match?` will convert to a regex (which will work fine, but is unnecessary)